isthmus/src/main/java/io/substrait/isthmus/SubstraitToSql.java

```diff
@@ -0,0 +1,85 @@
+package io.substrait.isthmus;
+
+import io.substrait.relation.Rel;
+import java.util.Arrays;
+import java.util.List;
+import java.util.function.UnaryOperator;
+import org.apache.calcite.config.CalciteConnectionConfig;
+import org.apache.calcite.config.CalciteConnectionProperty;
+import org.apache.calcite.jdbc.CalciteSchema;
+import org.apache.calcite.jdbc.JavaTypeFactoryImpl;
+import org.apache.calcite.plan.Contexts;
+import org.apache.calcite.plan.RelOptCluster;
+import org.apache.calcite.plan.RelOptCostImpl;
+import org.apache.calcite.plan.volcano.VolcanoPlanner;
+import org.apache.calcite.prepare.CalciteCatalogReader;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.rel2sql.RelToSqlConverter;
+import org.apache.calcite.rel.type.RelDataTypeFactory;
+import org.apache.calcite.rex.RexBuilder;
+import org.apache.calcite.sql.SqlDialect;
+import org.apache.calcite.sql.SqlNode;
+import org.apache.calcite.sql.SqlWriterConfig;
+import org.apache.calcite.sql.parser.SqlParseException;
+import org.apache.calcite.sql.validate.SqlValidator;
+import org.apache.calcite.sql2rel.SqlToRelConverter;
+
+public class SubstraitToSql extends SqlConverterBase {
+  private final RelDataTypeFactory factory;
+  private final CalciteCatalogReader catalogReader;
+  private final RelOptCluster relOptCluster;
+  private final SqlValidator validator;
+
+  SubstraitToSql() {
+    CalciteSchema rootSchema = CalciteSchema.createRootSchema(false);
+    SqlToRelConverter.Config converterConfig =
+        SqlToRelConverter.config().withTrimUnusedFields(true).withExpand(false);
+    this.factory = new JavaTypeFactoryImpl();
+
+    CalciteConnectionConfig config =
```

Comment:
This feels duplicative of other places. Can we factor out?

Reply:
I moved the duplicative code into SqlConverterBase.  The two class `SqlToSubstrait` and `SubstraitToSql` will share those common codes.